File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long sum = 0;
        long count = 0;
        for (int num: nums) {
            if (num == 0) {
                count++;
            } else if (num != 0 && count != 0) {
                sum += ((1+count) * count)/2;
                count = 0;
            }
        }
        if (count != 0) {
                sum += ((1+count) * count)/2;
                count = 0;
            }
        return sum;
    }
}

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        current_sum = 0
        total_sum = 0
        for num in nums:
            if num != 0:
                if current_sum != 0:
                    total_sum += (current_sum*(current_sum+1))//2
                current_sum = 0
            else:
                current_sum += 1
        if current_sum != 0:
            total_sum += (current_sum*(current_sum+1))//2
        return total_sum
        